flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer08.java

```diff
@@ -345,16 +343,19 @@ protected static void validateZooKeeperConfig(Properties props) {
 		}
 	}
 
-	private static long getInvalidOffsetBehavior(Properties config) {
+	/**
+	 * Check for invalid "auto.offset.reset" values. Should be called in constructor for eager checking before submitting
+	 * the job. Note that 'none' is also considered invalid, as we don't want to deliberately throw an exception
```

Comment:
Look out for https://issues.apache.org/jira/browse/KAFKA-3370  if you aren't already aware.  Right now allowing none and catching the exception only on startup is the best workaround I've seen.